__tests__/deleteUsers.test.js

```diff
@@ -0,0 +1,57 @@
+const name = "DeleteUsers";
+const logger = require("../logger");
+const { mockedAxiosClient } = require("../__mocks__/network");
+const formAxiosMock = require("../__mocks__/gen-axios.mock");
+const deleteUserDestinations = [
+  "am",
+  "braze",
+  "intercom",
+  "mp",
+  "af",
+  "clevertap",
+  "engage",
+  "ga"
+];
+const { handleDeletionOfUsers } = require("../versionedRouter");
+const { default: axios } = require("axios");
+
+
+
+// delete user tests
+deleteUserDestinations.forEach(destination => {
+  const inputData = require(`./data/${destination}_deleteUsers_proxy_input.json`);
+  const expectedData = require(`./data/${destination}_deleteUsers_proxy_output.json`);
+  
+  describe(`${name} Tests: ${destination}`, () => {
+    beforeAll(() => {
+      let axiosResponses;
+      try {
+        axiosResponses = require(`./data/${destination}_deleteUsers_response.json`);
+      } catch (error) {
+       // Do nothing
+       logger.error(`Error while reading ${destination}_deleteUsers_response.json: ${error}`)
+      }
+      if (Array.isArray(axiosResponses)) {
```

Comment:
the major part to test in delete flow is around forming the right req body & if that itself is mocked, what purpose are we trying to achieve for deleteUsers flow ?

Reply:
I see the need of having unit tests for this module.
We can use ga_deleteUsers_response.json to assert the reqBody generation in userDeletionHandler. And continue using ga_deleteUsers_response.json to mock for complete component test as happening now too.

Reply:
That was the case even previously & not in a right way. When no mock response is provided, the `mockedAxiosClient` used to return `{}`, which when processed would return `200` for destinations other than `GA`.

**Purpose**:
1. With this PR, we have provided a mechanism where in we have a way to mention the mock responses separately
2. Also we have implemented a generic axios mock client which can be used for any other requirements as well. Which I feel is a lot more extensible
3. Separating it from `proxy` tests made more sense as we might have other destinations which send a different response to deletion as opposed to event-delivery(as we see in `GA`)

Reply:
Moreover, this approach will fail if the requests are made parallelly.

Reply:
>  assert the reqBody generation in userDeletionHandler

The test-cases for the same has been added. Can you once check and confirm if that suffices ?

Reply:
Although the new changes you made succeeded in fulfilling the objective, I'm not too sure that's the right way. 
We cannot ensure all the user deletion handlers do not change or add something to the return value of prepare request function while making the request.

We should intercept the requests to Axios instead, if feasible.

Reply:
>We should intercept the requests to Axios instead

Can you elaborate ?

Reply:
Please look at the code mentioned in the conversation link
https://github.com/rudderlabs/rudder-transformer/pull/1551/files#r1018180720

Reply:
The fixes has been made in this PR. Can you take a look @saikumarrs ?
https://github.com/rudderlabs/rudder-transformer/pull/1563

Reply:
LGTM